test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -485,8 +484,6 @@ public void checkWarningHeaders(final List<String> warningHeaders, String testPa
             }
         }
 
-        unexpected.removeIf(s -> s.endsWith(SourceFieldMapper.DEPRECATION_WARNING + "\""));
-
```

Comment:
Remove logic that allows deprecation warning because of _source.mode mapping attribute usage.